docs/source/extending.rst

```diff
@@ -0,0 +1,144 @@
+* Leading and trailing slashes are stripped.  For example, ``/foo/bar/buzz/``
+  becomes ``foo/bar/buzz``.
+* The root directory is represented with the empty string.
+
+Filesystem Entities
+'''''''''''''''''''
+.. _notebook models:
+
+ContentsManager methods represent virtual filesystem entities as dictionaries,
+which we refer to as **models**.
+
+Models may contain the following entries:
+
++--------------------+-----------+------------------------------+
+| Key                | Type      |Info                          |
++====================+===========+==============================+
+|**name**            |unicode    |Basename of the entity.       |
++--------------------+-----------+------------------------------+
+|**path**            |unicode    |Full                          |
+|                    |           |(:ref:`API-style<apipaths>`)  |
+|                    |           |path to the entity.           |
++--------------------+-----------+------------------------------+
+|**type**            |unicode    |The entity type. One of       |
+|                    |           |``"notebook"``, ``"file"`` or |
+|                    |           |``"directory"``.              |
++--------------------+-----------+------------------------------+
+|**created**         |unicode    |Creation date of the entity,  |
+|                    |           |as ISO-8601 string.           |
+|                    |           |                              |
++--------------------+-----------+------------------------------+
+|**last_modified**   |unicode    |Last modified date of the     |
+|                    |           |entity, as ISO-8601 string.   |
+|                    |           |                              |
++--------------------+-----------+------------------------------+
+|**content**         |variable   |The "content" of the entity.  |
+|                    |           |(:ref:`See                    |
+|                    |           |Below<modelcontent>`)         |
++--------------------+-----------+------------------------------+
+|**mimetype**        |unicode or |The mimetype of ``"content"``,|
+|                    |``None``   |if any.  (:ref:`See           |
+|                    |           |Below<modelcontent>`)         |
++--------------------+-----------+------------------------------+
+|**format**          |unicode or |The format of ``"content"``,  |
+|                    |``None``   |if any. (:ref:`See            |
+|                    |           |Below<modelcontent>`)         |
++--------------------+-----------+------------------------------+
+
+.. _modelcontent:
+
+The **"content"** field
+
+In certain circumstances, we may not need the full content of an entity to
+complete a Contents API request.  require the full content of a filesystem
+entity.  For example, if we want to list the entries in directory ""foo/bar",
+we don't want to include the data stored in each file
```

Comment:
This section should probably also mention the types that content can have when it is used, and the significance of the `mimetype` and `format` fields.